api_client.go

```diff
@@ -0,0 +1,1031 @@
+            headerParams["Content-Type"] = contentType
+        }
+
+        body, err = setBody(postBody, contentType)
+        if err != nil {
+            return nil, err
+        }
+    }
+
+    // add form parameters and file if available.
+    if strings.HasPrefix(headerParams["Content-Type"], "multipart/form-data") && len(formParams) > 0 || (len(fileBytes) > 0 && fileName != "") {
+        if body != nil {
+            return nil, errors.New("Cannot specify postBody and multipart form at the same time.")
+        }
+        body = &bytes.Buffer{}
+        w := multipart.NewWriter(body)
+
+        for k, v := range formParams {
+            for _, iv := range v {
+                if strings.HasPrefix(k, "@") { // file
+                    err = addFile(w, k[1:], iv)
+                    if err != nil {
+                        return nil, err
+                    }
+                } else { // form value
+                    w.WriteField(k, iv)
+                }
+            }
+        }
+        if len(fileBytes) > 0 && fileName != "" {
+            w.Boundary()
+            //_, fileNm := filepath.Split(fileName)
+            part, err := w.CreateFormFile(formFileName, filepath.Base(fileName))
+            if err != nil {
+                return nil, err
+            }
+            _, err = part.Write(fileBytes)
+            if err != nil {
+                return nil, err
+            }
+        }
+
+        // Set the Boundary in the Content-Type
+        headerParams["Content-Type"] = w.FormDataContentType()
+
+        // Set Content-Length
+        headerParams["Content-Length"] = fmt.Sprintf("%d", body.Len())
+        w.Close()
+    }
+
+    if strings.HasPrefix(headerParams["Content-Type"], "application/x-www-form-urlencoded") && len(formParams) > 0 {
+        if body != nil {
+            return nil, errors.New("Cannot specify postBody and x-www-form-urlencoded form at the same time.")
+        }
+        body = &bytes.Buffer{}
+        body.WriteString(formParams.Encode())
+        // Set Content-Length
+        headerParams["Content-Length"] = fmt.Sprintf("%d", body.Len())
+    }
+
+    // Setup path and query parameters
+    url, err := url.Parse(path)
+    if err != nil {
+        return nil, err
+    }
+
+    // Override request host, if applicable
+    if c.cfg.ApiHost != "" {
+        url.Host = c.cfg.ApiHost
+    }
+
+    // Override request scheme, if applicable
+    if c.cfg.ApiScheme != "" {
+        url.Scheme = c.cfg.ApiScheme
+    }
+
+    // Adding Query Param
+    query := url.Query()
+    for k, v := range queryParams {
+        for _, iv := range v {
+            query.Add(k, iv)
+        }
+    }
+
+    // Encode the parameters.
+    url.RawQuery = query.Encode()
+
+    // Generate a new request
+    if body != nil {
+        localVarRequest, err = http.NewRequest(method, url.String(), body)
+    } else {
+        localVarRequest, err = http.NewRequest(method, url.String(), nil)
```

Comment:
this is weird ...

Reply:
autogenerated, but will modify it

Reply:
This cannot be changed - causes null pointer exceptions downstream